PayloadPkg/FirmwareUpdate/GetCapsuleImage.c

```diff
@@ -36,42 +38,46 @@
 /**
   Get hardware partition handle from boot option info
 
-  This function will initialize boot device, and get hardware partition
+  This function will initialize boot device and get hardware partition
   handle based on boot option.
 
-  @param[in]  BootOption      Current boot option
+  @param[in]  CapsuleInfo     Pointer to capsule information config data
   @param[out] HwPartHandle    Hardware partition handle for boot image
 
   @retval  RETURN_SUCCESS     If partition was found successfully
   @retval  Others             If partition was not found
 **/
 EFI_STATUS
 FindBootPartition (
-  IN  OS_BOOT_OPTION         *BootOption,
-  OUT EFI_HANDLE             *HwPartHandle
+  IN  CAPSULE_INFO_CFG_DATA *CapsuleInfo,
+  OUT EFI_HANDLE            *HwPartHandle
   )
 {
-  RETURN_STATUS              Status;
-  UINTN                      BootMediumPciBase;
+  RETURN_STATUS   Status;
+  UINTN           BootMediumPciBase;
+
+  if (CapsuleInfo == NULL) {
+    return EFI_INVALID_PARAMETER;
+  }
 
   //
   // Get OS boot device address
   //
-  if ((BootOption->DevAddr & 0xFF000000) == 0) {
+  if ((CapsuleInfo->DevAddr & 0xFF000000) == 0) {
```

Comment:
There is new macro for it.   Can you align with  TO_MM_PCI_ADDRESS() ?

Reply:
Yes, Please make sure your patch is based on latest code, since I already update the file to use TO_MM_PCI_ADDRESS().

Reply:
Updated the code to latest